core/src/main/resources/hudson/model/View/index.jelly

```diff
@@ -44,11 +44,24 @@ THE SOFTWARE.
         <l:header>
             <!-- for screen resolution detection -->
             <l:yui module="cookie" />
-            <script>
-              YAHOO.util.Cookie.set("screenResolution", screen.width+"x"+screen.height, {
-                path: "/"
-              });
-            </script>
+            <j:invokeStatic var="req" className="org.kohsuke.stapler.Stapler" method="getCurrentRequest"/>
+            <j:choose>
+              <j:when test="${req.isSecure()}">
+                <script>
+                  YAHOO.util.Cookie.set("screenResolution", screen.width+"x"+screen.height, {
+                    path: "/",
+                    secure: true
```

Comment:
Could we not just have one branch?
```suggestion
                    secure: ${req.secure}
```
or similar?

Reply:
we could and I am happy to do that.  I got the impression @Wadeck was not in favour - but as this is just a boolean it should be completely safe

Reply:
both works, for CSP we will eventually move it to its own file ;)

Reply:
> for CSP we will eventually move it to its own file

in which case we would need 2 files IIUC, or the javascript would need to work out if the endpoint would be https and then do the right thing?

would it be better to keep if choose/when/then  because of that?

happy to change to whatever the consensus is with @jglick

Reply:
💡 By using the technique from https://www.jenkins.io/doc/developer/security/xss-prevention/#passing-values-to-javascript, you can pass the variable as an attribute, that will be read by the javascript. With this approach, you can have a single JavaScript file that will behave differently according to the `secure`.
(this is out of the scope for this PR, I don't want to force you doing that just for this secure cookie :D)

Reply:
> whatever the consensus is with @jglick

No strong opinion.